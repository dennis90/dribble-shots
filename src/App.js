import React, { Component } from 'react';
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'
import {HashRouter} from 'react-router-dom'

import Header from './components/header'
import ShotList from './components/Listing'
import * as actionCreators from './actionCreators'


export class App extends Component {
  componentDidMount() {
    this.props.actions.fetchDribbleShots()
  }

  render() {
    return (
      <HashRouter>
        <div className="App">
          <Header />
          <ShotList {...this.props} />
        </div>
      </HashRouter>
    )
  }
}

const mapStateToProps = state => ({...state})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actionCreators, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
