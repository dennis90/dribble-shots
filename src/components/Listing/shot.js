import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
