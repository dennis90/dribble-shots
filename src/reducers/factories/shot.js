import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
