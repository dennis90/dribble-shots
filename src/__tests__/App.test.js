import React from 'react'
import {mount, shallow} from 'enzyme'
import {Factory} from 'rosie'
import {OrderedMap} from 'immutable'

import {App} from '../App'
import {ShotFactory} from '../reducers/factories/shot'


describe('App Component', () => {
  it('should render shot components', () => {
    const shots = ShotFactory.buildList(3).reduce((list, shot) => {
      return list.set(shot.id, shot)
    }, OrderedMap())

    const props = {
      shots,
      actions: {
        fetchDribbleShots: jest.fn()
      }
    }

    const wrapper = shallow(<App {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call `fetchDribbleShots` action creator after component mounted', () => {
    const shots = ShotFactory.buildList(3).reduce((list, shot) => {
      return list.set(shot.id, shot)
    }, OrderedMap())

    const props = {
      shots,
      actions: {
        fetchDribbleShots: jest.fn()
      }
    }

    const wrapper = shallow(<App {...props} />)
    expect(props.actions.fetchDribbleShots).toBeCalled()
  })
})
