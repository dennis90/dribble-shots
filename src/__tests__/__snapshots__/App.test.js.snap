// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render shot components 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    actions={
        Object {
            "fetchDribbleShots": [Function],
          }
    }
    shots={
        Immutable.OrderedMap {
            1: Object {
                "description": "Shot Description #1",
                "id": 1,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #1",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #1",
                },
                "views_count": 100,
              },
            2: Object {
                "description": "Shot Description #2",
                "id": 2,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #2",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #2",
                },
                "views_count": 100,
              },
            3: Object {
                "description": "Shot Description #3",
                "id": 3,
                "images": Object {
                  "hidpi": "https://dribbble/images/hidpi",
                  "normal": "https://dribbble/images/normal",
                },
                "likes_count": 100,
                "title": "Shot #3",
                "user": Object {
                  "avatar_url": "https://dribbble.com/user/avatar.png",
                  "html_url": "https://dribbble.com/user/",
                  "name": "User #3",
                },
                "views_count": 100,
              },
        }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <Header />
        <Unknown
                actions={
                        Object {
                                "fetchDribbleShots": [Function],
                              }
                }
                shots={
                        Immutable.OrderedMap {
                                1: Object {
                                        "description": "Shot Description #1",
                                        "id": 1,
                                        "images": Object {
                                          "hidpi": "https://dribbble/images/hidpi",
                                          "normal": "https://dribbble/images/normal",
                                        },
                                        "likes_count": 100,
                                        "title": "Shot #1",
                                        "user": Object {
                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                          "html_url": "https://dribbble.com/user/",
                                          "name": "User #1",
                                        },
                                        "views_count": 100,
                                      },
                                2: Object {
                                        "description": "Shot Description #2",
                                        "id": 2,
                                        "images": Object {
                                          "hidpi": "https://dribbble/images/hidpi",
                                          "normal": "https://dribbble/images/normal",
                                        },
                                        "likes_count": 100,
                                        "title": "Shot #2",
                                        "user": Object {
                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                          "html_url": "https://dribbble.com/user/",
                                          "name": "User #2",
                                        },
                                        "views_count": 100,
                                      },
                                3: Object {
                                        "description": "Shot Description #3",
                                        "id": 3,
                                        "images": Object {
                                          "hidpi": "https://dribbble/images/hidpi",
                                          "normal": "https://dribbble/images/normal",
                                        },
                                        "likes_count": 100,
                                        "title": "Shot #3",
                                        "user": Object {
                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                          "html_url": "https://dribbble.com/user/",
                                          "name": "User #3",
                                        },
                                        "views_count": 100,
                                      },
                        }
                }
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Unknown
            actions={
                        Object {
                                    "fetchDribbleShots": [Function],
                                  }
            }
            shots={
                        Immutable.OrderedMap {
                                    1: Object {
                                                "description": "Shot Description #1",
                                                "id": 1,
                                                "images": Object {
                                                  "hidpi": "https://dribbble/images/hidpi",
                                                  "normal": "https://dribbble/images/normal",
                                                },
                                                "likes_count": 100,
                                                "title": "Shot #1",
                                                "user": Object {
                                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                                  "html_url": "https://dribbble.com/user/",
                                                  "name": "User #1",
                                                },
                                                "views_count": 100,
                                              },
                                    2: Object {
                                                "description": "Shot Description #2",
                                                "id": 2,
                                                "images": Object {
                                                  "hidpi": "https://dribbble/images/hidpi",
                                                  "normal": "https://dribbble/images/normal",
                                                },
                                                "likes_count": 100,
                                                "title": "Shot #2",
                                                "user": Object {
                                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                                  "html_url": "https://dribbble.com/user/",
                                                  "name": "User #2",
                                                },
                                                "views_count": 100,
                                              },
                                    3: Object {
                                                "description": "Shot Description #3",
                                                "id": 3,
                                                "images": Object {
                                                  "hidpi": "https://dribbble/images/hidpi",
                                                  "normal": "https://dribbble/images/normal",
                                                },
                                                "likes_count": 100,
                                                "title": "Shot #3",
                                                "user": Object {
                                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                                  "html_url": "https://dribbble.com/user/",
                                                  "name": "User #3",
                                                },
                                                "views_count": 100,
                                              },
                        }
            }
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "fetchDribbleShots": [Function],
            },
            "shots": Immutable.OrderedMap {
              1: Object {
                            "description": "Shot Description #1",
                            "id": 1,
                            "images": Object {
                              "hidpi": "https://dribbble/images/hidpi",
                              "normal": "https://dribbble/images/normal",
                            },
                            "likes_count": 100,
                            "title": "Shot #1",
                            "user": Object {
                              "avatar_url": "https://dribbble.com/user/avatar.png",
                              "html_url": "https://dribbble.com/user/",
                              "name": "User #1",
                            },
                            "views_count": 100,
                          },
              2: Object {
                            "description": "Shot Description #2",
                            "id": 2,
                            "images": Object {
                              "hidpi": "https://dribbble/images/hidpi",
                              "normal": "https://dribbble/images/normal",
                            },
                            "likes_count": 100,
                            "title": "Shot #2",
                            "user": Object {
                              "avatar_url": "https://dribbble.com/user/avatar.png",
                              "html_url": "https://dribbble.com/user/",
                              "name": "User #2",
                            },
                            "views_count": 100,
                          },
              3: Object {
                            "description": "Shot Description #3",
                            "id": 3,
                            "images": Object {
                              "hidpi": "https://dribbble/images/hidpi",
                              "normal": "https://dribbble/images/normal",
                            },
                            "likes_count": 100,
                            "title": "Shot #3",
                            "user": Object {
                              "avatar_url": "https://dribbble.com/user/avatar.png",
                              "html_url": "https://dribbble.com/user/",
                              "name": "User #3",
                            },
                            "views_count": 100,
                          },
},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <Header />
          <Unknown
                    actions={
                              Object {
                                        "fetchDribbleShots": [Function],
                                      }
                    }
                    shots={
                              Immutable.OrderedMap {
                                        1: Object {
                                                  "description": "Shot Description #1",
                                                  "id": 1,
                                                  "images": Object {
                                                    "hidpi": "https://dribbble/images/hidpi",
                                                    "normal": "https://dribbble/images/normal",
                                                  },
                                                  "likes_count": 100,
                                                  "title": "Shot #1",
                                                  "user": Object {
                                                    "avatar_url": "https://dribbble.com/user/avatar.png",
                                                    "html_url": "https://dribbble.com/user/",
                                                    "name": "User #1",
                                                  },
                                                  "views_count": 100,
                                                },
                                        2: Object {
                                                  "description": "Shot Description #2",
                                                  "id": 2,
                                                  "images": Object {
                                                    "hidpi": "https://dribbble/images/hidpi",
                                                    "normal": "https://dribbble/images/normal",
                                                  },
                                                  "likes_count": 100,
                                                  "title": "Shot #2",
                                                  "user": Object {
                                                    "avatar_url": "https://dribbble.com/user/avatar.png",
                                                    "html_url": "https://dribbble.com/user/",
                                                    "name": "User #2",
                                                  },
                                                  "views_count": 100,
                                                },
                                        3: Object {
                                                  "description": "Shot Description #3",
                                                  "id": 3,
                                                  "images": Object {
                                                    "hidpi": "https://dribbble/images/hidpi",
                                                    "normal": "https://dribbble/images/normal",
                                                  },
                                                  "likes_count": 100,
                                                  "title": "Shot #3",
                                                  "user": Object {
                                                    "avatar_url": "https://dribbble.com/user/avatar.png",
                                                    "html_url": "https://dribbble.com/user/",
                                                    "name": "User #3",
                                                  },
                                                  "views_count": 100,
                                                },
                              }
                    }
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header />,
            <Unknown
              actions={
                            Object {
                                          "fetchDribbleShots": [Function],
                                        }
              }
              shots={
                            Immutable.OrderedMap {
                                          1: Object {
                                                        "description": "Shot Description #1",
                                                        "id": 1,
                                                        "images": Object {
                                                          "hidpi": "https://dribbble/images/hidpi",
                                                          "normal": "https://dribbble/images/normal",
                                                        },
                                                        "likes_count": 100,
                                                        "title": "Shot #1",
                                                        "user": Object {
                                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                                          "html_url": "https://dribbble.com/user/",
                                                          "name": "User #1",
                                                        },
                                                        "views_count": 100,
                                                      },
                                          2: Object {
                                                        "description": "Shot Description #2",
                                                        "id": 2,
                                                        "images": Object {
                                                          "hidpi": "https://dribbble/images/hidpi",
                                                          "normal": "https://dribbble/images/normal",
                                                        },
                                                        "likes_count": 100,
                                                        "title": "Shot #2",
                                                        "user": Object {
                                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                                          "html_url": "https://dribbble.com/user/",
                                                          "name": "User #2",
                                                        },
                                                        "views_count": 100,
                                                      },
                                          3: Object {
                                                        "description": "Shot Description #3",
                                                        "id": 3,
                                                        "images": Object {
                                                          "hidpi": "https://dribbble/images/hidpi",
                                                          "normal": "https://dribbble/images/normal",
                                                        },
                                                        "likes_count": 100,
                                                        "title": "Shot #3",
                                                        "user": Object {
                                                          "avatar_url": "https://dribbble.com/user/avatar.png",
                                                          "html_url": "https://dribbble.com/user/",
                                                          "name": "User #3",
                                                        },
                                                        "views_count": 100,
                                                      },
                            }
              }
/>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "actions": Object {
                "fetchDribbleShots": [Function],
              },
              "shots": Immutable.OrderedMap {
                1: Object {
                                "description": "Shot Description #1",
                                "id": 1,
                                "images": Object {
                                  "hidpi": "https://dribbble/images/hidpi",
                                  "normal": "https://dribbble/images/normal",
                                },
                                "likes_count": 100,
                                "title": "Shot #1",
                                "user": Object {
                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                  "html_url": "https://dribbble.com/user/",
                                  "name": "User #1",
                                },
                                "views_count": 100,
                              },
                2: Object {
                                "description": "Shot Description #2",
                                "id": 2,
                                "images": Object {
                                  "hidpi": "https://dribbble/images/hidpi",
                                  "normal": "https://dribbble/images/normal",
                                },
                                "likes_count": 100,
                                "title": "Shot #2",
                                "user": Object {
                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                  "html_url": "https://dribbble.com/user/",
                                  "name": "User #2",
                                },
                                "views_count": 100,
                              },
                3: Object {
                                "description": "Shot Description #3",
                                "id": 3,
                                "images": Object {
                                  "hidpi": "https://dribbble/images/hidpi",
                                  "normal": "https://dribbble/images/normal",
                                },
                                "likes_count": 100,
                                "title": "Shot #3",
                                "user": Object {
                                  "avatar_url": "https://dribbble.com/user/avatar.png",
                                  "html_url": "https://dribbble.com/user/",
                                  "name": "User #3",
                                },
                                "views_count": 100,
                              },
},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
